Add ERC: Dynamic Identity Binding Soulbound Tokens (#260)
ethereum/ERCs: ERCS/erc-7624.md

---
eip: 7624
title: Dynamic Identity Binding Soulbound Tokens

> description: Dynamic identity binding for SBT to provide a more resilient and user-centric solution

Reviewer: ```suggestion
description: Dynamic identity binding for Soulbound Tokens (SBT) to provide a more resilient and user-centric solution
```

author: Auph (@auph)
discussions-to: https://ethereum-magicians.org/t/erc-7624-dynamic-identity-binding-soulbound-tokens/18676
status: Draft
type: Standards Track
category: ERC
created: 2022-10-18
requires: 721
---

## Abstract

> This Ethereum Improvement Proposal introduces a mechanism for dynamic identity binding to enhance the security and continuity of users' identities in the context of Soulbound tokens. The proposal addresses the current limitation where Soulbound tokens are static and irrevocable, making it challenging for users to recover their identity in case of wallet compromise. The proposed solution allows users to re-identify themselves with a new wallet address while preserving the integrity of their Soulbound identity.

Reviewer: ```suggestion
This EIP introduces a mechanism for dynamic identity binding to enhance the security and continuity of users' identities in the context of Soulbound tokens. The proposal addresses the current limitation where Soulbound tokens are static and irrevocable, making it challenging for users to recover their identity in case of wallet compromise. The proposed solution allows users to re-identify themselves with a new wallet address while preserving the integrity of their Soulbound identity.
```

> This Ethereum Improvement Proposal introduces a mechanism for dynamic identity binding to enhance the security and continuity of users' identities in the context of Soulbound tokens. The proposal addresses the current limitation where Soulbound tokens are static and irrevocable, making it challenging for users to recover their identity in case of wallet compromise. The proposed solution allows users to re-identify themselves with a new wallet address while preserving the integrity of their Soulbound identity.

Reviewer: Your Abstract is missing the "how". An EIP abstract should contain some technical information on how you will achieve the goal.

## Motivation

The existing Soulbound token system lacks the flexibility to adapt to compromised wallet scenarios, leaving users with no on-chain means to seamlessly transition to a new wallet address without losing their unique identity. Despite the availability of Account Abstraction (AA) - ERC-4337 for social recovery, residual risks persist, prompting the need for a more comprehensive solution that ensures the continuity of a user's identity.

## Specification

> **Soulbound Token Modification**

Reviewer: ```suggestion
### Soulbound Token Modification
```

**1. Dynamic Binding:**
- Introduce a mechanism for dynamic binding of Soulbound tokens to wallet addresses, allowing users to update their associated wallet address securely.
- This dynamic binding should be achievable through a secure on-chain process, ensuring that the user retains control over the migration of their Soulbound identity.
- This can include binding social links, recovery addresses (i.e. guardians), and security answers.

**2. Revocation Mechanism:**
- Implement a revocation mechanism that allows users to revoke the association of a Soulbound token with a compromised wallet address, rendering it inactive and preventing malicious use.

> **Authorization**

Reviewer: ```suggestion
### Authorization
```

**1. Secondary Private Key**

> - Implement a secondary private and public key for each address that will be used to authenticate Revocation (private key) and reveal metadata (public key) such as social links.

Reviewer: Is "Revocation", with a capital, a specialised term?

> **User Initiated Identity Migration**

Reviewer: ```suggestion
### User Initiated Identity Migration
```

**1. Identity Migration Contract:**
- Develop a smart contract that facilitates the secure migration of Soulbound identities from one wallet address to another, ensuring that the process is initiated and controlled by the rightful owner of the Soulbound token.

**2. Proof of Ownership:**
- Implement a secure method for users to provide proof of ownership of the Soulbound token associated with the compromised wallet address during the migration process.

**3. Gas-Efficient Migration:**
- Optimize the identity migration process to minimize gas costs, making it feasible for users to perform the migration without excessive transaction fees.

## Rationale

The proposed changes aim to provide a comprehensive solution for identity management on the Ethereum blockchain. By allowing dynamic identity binding and secure migration, users can maintain their unique Soulbound identity even in the face of security breaches, enhancing the overall security posture of Soulbound tokens.

> 

Reviewer: The Rationale section should explain the choices made in the EIP. This section is currently not explaining any particulars of the EIP.

> ## Implementation
> 
> The implementation of this proposal requires updates to existing Soulbound token contracts, introducing new functionalities for dynamic binding, revocation, user authorization and identity migration. A reference implementation and thorough testing are necessary to ensure the robustness and security of the proposed changes. A cryptographic function to derive unique private keys based on the token ID.
> 

Reviewer: ```suggestion
```

This isn't a valid section.

## Backwards Compatibility

> Efforts should be made to ensure backward compatibility with existing Soulbound token contracts, allowing for a smooth transition to the new dynamic identity binding mechanism.

Reviewer: Although not a formatting issue, this section is not explaining your compatibility issues. You should go more in depth.

## Security Considerations

**1. Secure Private Key Handling:**

Careful consideration must be given to the handling of private keys, ensuring they are securely managed and stored to prevent unauthorized access and misuse.

**2. Immutable Data Hashing:**

All sensitive data hashed within dSBT NFTs must be immutable and tamper-proof, ensuring the integrity and confidentiality of user information.

**3. Authentication Mechanisms:**

Robust authentication mechanisms must be implemented to verify the identity of users initiating token revocations or identity migrations, preventing unauthorized actions.

**4. Gas-Efficient Design:**

> Gas-efficient design practices should be employed to minimize transaction costs associated with identity migrations and token revocations, ensuring affordability for users.

Reviewer: Not a security consideration.

> ## Conclusion
> 
> The introduction of dynamic identity binding for Soulbound tokens provides a more resilient and user-centric solution, addressing the limitations of the current static and irrevocable system. This proposal aims to enhance the overall security and user experience of Soulbound tokens on the Ethereum blockchain. Feedback from the Ethereum community is encouraged for further refinement before implementation.

Reviewer: ```suggestion
```

Not a valid section.
